sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -102,9 +106,9 @@ class MapboxMap internal constructor(
   ) {
     initializeStyleLoad(onStyleLoaded, onMapLoadErrorListener)
     if (styleUri.isEmpty()) {
-      nativeMapWeakRef.call { (this as StyleManagerInterface).styleJSON = EMPTY_STYLE_JSON }
+      nativeMapWeakRef.call { styleJSON = EMPTY_STYLE_JSON }
```

Comment:
Recommend to use `this` keyword to limit the scope of the function to be within nativeMap, so that not to be confused with  property/functions with the same name in the `MapboxMap` class. It used to cause trouble since we have the exact same functions both inside the `NativeMap` and `MapboxMap`.
```suggestion
      nativeMapWeakRef.call { this.styleJSON = EMPTY_STYLE_JSON }
```

Reply:
Could you explain a bit more? `this.` is implied anyway, 
`nativeMapWeakRef.call { someProperty ->`
should be always the same as 
`nativeMapWeakRef.call { this.someProperty`

Reply:
we have had some occurrences where it was calling the function on the outer class versus on the native map. Not much of an issue here though

Reply:
Ah, got it, it seems. 
IMO for such situations when `this.` context might be confused and outer class may hold similar or same method - better to use explicit named references with let e.g. `let { inner -> inner.smth }` instead of apply or similar that changes `this.`